rust/worker/Cargo.toml

```diff
@@ -47,6 +52,7 @@ proptest = "1.4.0"
 proptest-state-machine = "0.1.0"
 "rand" = "0.8.5"
 rayon = "1.8.0"
+criterion = "0.3"
```

Comment:
out of curiosity what alternatives did we evaluate?

Reply:
Good question. By default, we have `libtest` using which we can do basic benchmarking. My main motive behind using `criterion` was to enable sophisticated benchmarking for the future. In particular `criterion` provides:
1. Statistics: Statistical analysis detects if, and by how much, performance has changed since the last benchmark run
2. Charts: Uses [gnuplot](http://www.gnuplot.info/) to generate detailed graphs of benchmark results

These would be useful in future to run these benchmarks in our CI/CD pipeline or nightly/weekly (some cadence) to detect perf regressions for e.g.

Reply:
Awesome thank you